Sources/Subs-Db-postgresql.php

```diff
@@ -202,6 +202,13 @@ function smf_db_replacement__callback($matches)
 				smf_db_error_backtrace('Wrong value type sent to the database. Date expected. (' . $matches[2] . ')', '', E_USER_ERROR, __FILE__, __LINE__);
 		break;
 
+		case 'time':
+			if (preg_match('~^([0-1]?\d|2[0-3]):([0-5]\d):([0-5]\d)$~', $replacement, $time_matches) === 1)
+				return sprintf('\'%02d:%02d:%02d\'', $time_matches[1], $time_matches[2], $time_matches[3]);
```

Comment:
Sure otherwise is it "luke" That the rdbms reconize as time and in the right format.


________________________________
Von: Jon Stovell <notifications@github.com>
Gesendet: 31.07.2016 20:53
An: SimpleMachines/SMF2.1
Cc: albertlast; Comment
Betreff: Re: [SimpleMachines/SMF2.1] The calendar now supports event times (#3530)

> @@ -202,6 +202,13 @@ function smf_db_replacement__callback($matches)
>                               smf_db_error_backtrace('Wrong value type sent to the database. Date expected. (' . $matches[2] . ')', '', E_USER_ERROR, __FILE__, __LINE__);
>               break;
>
> +             case 'time':
> +                     if (preg_match('~^([0-1]?\d|2[0-3]):([0-5]\d):([0-5]\d)$~', $replacement, $time_matches) === 1)
> +                             return sprintf('\'%02d:%02d:%02d\'', $time_matches[1], $time_matches[2], $time_matches[3]);

... Or do you mean that instead of returning a string like `15:15:20` that it should instead return `TO_TIMESTAMP('15:15:20', 'HH24:MI:SS')::TIME`?

---
You are receiving this because you commented.
Reply to this email directly or view it on GitHub:
https://github.com/SimpleMachines/SMF2.1/pull/3530/files/b7128f146b6bf1a9673095524c7ff7f004466d1c#r72908810